### PR TITLE
Remove patch versions from in-files

### LINKED ISSUE #711: Remove patch versions from in-files
For most of our dependencies in the `-in` files we have major.minor version, but for a few we have major.minor.patch-version. We should remove the patch-version since they are not needed.

### PR DESCRIPTION
**Pull request by EinarJohnsen**
#Closes #711

### CODE REVIEW

File: dev_requirements.in [hunk 1]
```diff
@@ -1,9 +1,9 @@
 -c requirements.txt
```
Comment (flikka) [inline dev_requirements.in:1]: Is this line supposed to be here? Not your doing, but just looked weird.

File: requirements.in [hunk 2]
```diff
@@ -2,15 +2,15 @@ apscheduler~=3.6
...
+gunicorn~=20.0
 h5py~=2.8
 influxdb~=5.2
 jinja2~=2.10
 pyarrow~=0.15
 numpy~=1.17
 pandas~=0.25
 peewee~=3.12
-numexpr>=2.6.1
+numexpr>=2.6
```
Comment (flikka) [inline requirements.in]: I wonder if we have a conscious and predictable approach to using compatible (~=) vs greaterequal (>=). To me, it feels sensible to use greater than unless we are "afraid" greater than will break our stuff, makes it easier to live together with other requirements (?). But I don't know. Maybe someone else have opinions? @epa095 @milesgranger @ryanjdillon

File: requirements.in [hunk 3]
```diff
@@ -2,15 +2,15 @@ apscheduler~=3.6
 Click~=7.0
 cachetools~=3.1
 dictdiffer
```
Comment (flikka) [inline requirements.in]: missing version?

**Review by flikka — COMMENTED** [review]
Looks good, with a couple of comments/questions.

File: dev_requirements.in (hunk 1 above)
Comment (EinarJohnsen) [reply dev_requirements.in:1]: This looks like it is `pip-compile` specific and used to constrain the `dev_requirements.in` to packages used for "production" (as they call it in the example) in `requirements.txt`, so they are compatible, yet different.
There is an example towards the end under: "Workflow for layered requirements"
https://pypi.org/project/pip-tools/

File: requirements.in (hunk 2 above)
Comment (epa095) [reply requirements.in]: I prefer compatible with *as the general rule*. It only really makes a difference with major versions (thats where they differ), and with those I think it makes sense that we by default dont assume that we are compatible with newer major versions.

Comment (milesgranger) [reply requirements.in]: It does appear from looking at things like [sklearn's setup.py](https://github.com/scikit-learn/scikit-learn/blob/f4da713c44d9529da2d82d76e0c0c6ef7e0e4795/setup.py#L260) as well as [Dask's](https://github.com/dask/dask/blob/3ba4f6e43ecda75ed4ad19f42f3eba6c78a42d49/setup.py#L23), [requests](https://github.com/psf/requests/blob/eedd67462819f8dbf8c1c32e77f9070606605231/setup.py#L44) and others, that using `>=` seems to be the norm, even with requirements which aren't at a major version. :man_shrugging: 

But to be _truly_ accurate in this, we'd have to find a reasonable minimum for each lib. :sweat_smile:

File: requirements.in (hunk 3 above)
Comment (EinarJohnsen) [reply requirements.in]: Yeah, I see this. Should we set it `~=0.8`, as it is `dictdiffer==0.8.0` in `requirements.txt`
@epa095

**Comment by flikka** [comment]
I am thinking: For a library, using >= would be best, to make oneselves as "compatible as possible", and rather take the risk of breaking on too new versions. 

But for our "system", it is more important to be "stable", than to be a nice co-habitant with others, therefore maybe ~= is a good rule of thumb.

**Review by epa095 — APPROVED** [review]
LGTM

**Comment by EinarJohnsen** [comment]
Edited the pyyaml to be `pyyaml ~= 5.1`, and it worked with the test project, so I assume that is fine wine. @flikka

**Pull request merged**